packages/client/modules/admin/containers/Graphql/GraphqlContainer.tsx

```diff
@@ -19,19 +19,50 @@ const GQL = styled('div')({
 
 type SchemaType = 'Public' | 'Private'
 
+const safeParseJSON = (str: string | null) => {
+  if (!str) return null
+  try {
+    return JSON.parse(str)
+  } catch {
+    return null
+  }
+}
+
+const persistSchemaForTab = (currentSchema: SchemaType) => {
+  // get active tab idx
+  const tabStateStr = window.localStorage.getItem('graphiql:tabState')
+  if (!tabStateStr) return
+  const parsedTabState = safeParseJSON(tabStateStr)
+  if (!parsedTabState) return
+  const activeTabIdx = parsedTabState?.activeTabIndex ?? null
+  if (activeTabIdx === null) return
+  // set schema for the given tab
+  const tabSchemaLookupStr = window.localStorage.getItem('graphiql:tabSchemaLookup')
+  const parsedTabSchemaLookup = safeParseJSON(tabSchemaLookupStr)
+  const tabSchemaLookup = Array.isArray(parsedTabSchemaLookup) ? parsedTabSchemaLookup : []
+  tabSchemaLookup[activeTabIdx] = currentSchema
+  window.localStorage.setItem('graphiql:tabSchemaLookup', JSON.stringify(tabSchemaLookup))
+}
+
 const GraphqlContainer = () => {
   const [currentSchema, setCurrentSchema] = useState<SchemaType>(() => {
     return (window.localStorage.getItem(LocalStorageKey.GRAPHIQL_SCHEMA) as SchemaType) || 'Public'
   })
-
+  const introspectionResultRef = useRef({Public: '', Private: ''})
```

Comment:
changing the schema should not send a new introspection query. this caches them as they are fetched